subprojects/packagefiles/fluidsynth/meson.build

```diff
@@ -9,6 +9,33 @@ cc = meson.get_compiler('c')
 summary('Build type',     get_option('buildtype'), section : 'Build Summary')
 summary('Install prefix', get_option('prefix'),    section : 'Build Summary')
 
+# Due to symbol visibility issues, don't allow building 'both' library
+# types on MSVC
+if get_option('default_library') == 'both' and cc.get_id() == 'msvc'
+  error('default_library=both not supported when compiling with MSVC')
+endif
+
+# Is a shared library being built?
+is_shared_lib = get_option('default_library') == 'shared'
+
+# Set global WIN32 define if windows
+if host_machine.system() == 'windows'
+  add_project_arguments('-DWIN32=1', language : ['c', 'cpp'])
+endif
+
+# Detect if compiling in an MSYS2-based MinGW/Clang environment
+host_is_msys2 = false
+if host_machine.system() == 'windows'
+  message('Checking if Windows host is MSYS2 based')
+  shell_cmd = find_program('sh', required : false)
+  if shell_cmd.found() 
+    r = run_command('sh', '-c', '[ ! -z "$MSYSTEM" ] && [ "$MSYSTEM" != "MSYS" ] && exit 0; exit 1')
+    if r.returncode() == 0
+      message('MSYS2 based environment detected')
+      host_is_msys2 = true
+    endif
+  endif
+endif
```

Comment:
I debated on whether to use `sh` or `python` for this.

Ended up using `sh`, because it's guaranteed to exist on MSYS2, so therefore if it can't be found, the environment is definitely not MSYS2.

Reply:
All the messages here seem not very useful to the person seeing it on the console.

Consider:

```meson
shell_cmd = find_program('sh', required : false, disabler: true)
host_is_msys2 = run_command('sh', '-c', '[ ! -z "$MSYSTEM" ] && [ "$MSYSTEM" != "MSYS" ] && exit 0; exit 1', check: false).returncode() == 0
```

If the program is not found, it turns into a disabler object and any code that tries to evaluate it becomes a no-op (including the later branch `ext_deps += [dependency('intl' , static : true)]`)

Reply:
That simplifies things a lot. Thanks for the suggestion.

New implementation still hides the check behind a 'windows' check, just in case some *nix based system had defined `MSYSTEM` for some reason.